### PR TITLE
DRILL-7911 Use TestContainers-MySQL instead of wix-embedded-mysql

### PR DESCRIPTION
**Pull request by martin-g**
# [DRILL-7911](https://issues.apache.org/jira/browse/DRILL-7911): Use TestContainers-MySQL instead of wix-embedded-mysql

## Description

Apache Drill build fails on Linux ARM64 because Wix Embedded MySQL does not support ARM64 and this library is no more maintained.
This PR replaces Wix Embedded MySQL with TestContainers-MySQL. In addition the PR sets Maven dependency management for JNA and Netty which support Linux ARM64.

## Documentation
There are no user visible changes. The changes are only in the tests.

## Testing
org.apache.drill.exec.store.jdbc.TestJdbcPluginWithMySQLIT has been updated to use TestContainers to run MySQL/MariaDB.

### CODE REVIEW

File: contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithMySQLIT.java
Comment (luocooong) [inline contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithMySQLIT.java:64]: Could you please defined the version of db in the final variable?

**Review by luocooong — CHANGES_REQUESTED** [review]
@martin-g Thanks for the PR. There are minor things...

File: pom.xml
Comment (martin-g) [reply pom.xml:113]: JNA is an alternative of JNI that is used by other libraries to communicate with native libraries.
There has been an improvement to JNA (and transitively to TestContainers) via https://github.com/testcontainers/testcontainers-java/issues/3610

File: pom.xml
Comment (martin-g) [reply pom.xml:117]: I'll revert it. There is nothing aarch64 related in this version bump. The aarch64 improvements are only in netty-tcnative only.

File: contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithMySQLIT.java
Comment (martin-g) [reply contrib/storage-jdbc/src/test/java/org/apache/drill/exec/store/jdbc/TestJdbcPluginWithMySQLIT.java]: Done!

File: pom.xml
Comment (martin-g) [reply pom.xml:1168]: JNA comes as a transitive dependency of:
- org.kohsuke:libpam4j in drill-java-exec and drill-jdbc-all
- testcontainers/docker-java-api in all modules which use TestContainers (i.e. MySQL, Mongo, Splunk, Cassandra and Java Exec

I think defining it in the root pom is the best!

**Comment by luocooong** [comment]
@martin-g Thanks for the revision. Because the ARM-based CI is not ready. At the same time, we are going to release the 1.19. So, Would you mind merging your PR (related the ARM) in next release?

**Comment by martin-g** [comment]
Sure!
No problem!

**Comment by cgivre** [comment]
@luocooong 
Are we good to merge this?

**Comment by luocooong** [comment]
@cgivre I recommend that merge this PR after the DRILL-7923.

File: pom.xml
Comment (martin-g) [reply pom.xml:113]: Here is a build at Travis that fails due to the old JNA version in that branch:  https://travis-ci.com/github/apache/drill/jobs/504592762

**Comment by martin-g** [comment]
> I recommend that merge this PR after the DRILL-7923.

As I said at https://github.com/apache/drill/pull/2220#issue-637610515 : `This PR depends on #2217, #2218 and #2219. The build will fail before those are merged.`

I.e. you could merge #2220 (DRILL-7923) first but the build at Travis will fail until this PR is merged too.

**Comment by luocooong** [comment]
@martin-g Please squash all the commits (rebase on the mater).

**Comment by martin-g** [comment]
Squashed and rebased!

**Review by luocooong — APPROVED** [review]
@martin-g +1

File: pom.xml
Comment (martin-g) [reply pom.xml:455]: @luocooong Let's wait for a successful TravisCI build. It just failed due to some problem with TestContainers-Ryuk. I've updated TestContainers-Vault to 1.15.3.

This comment is documenting what is .cnf file. But I will remove it!

**Comment by martin-g** [comment]
The build has passed: 92759d96f258a1bcfd4eae33cf8f2410e406afbf
I've squashed the new commits.
I think it is ready to be merged!

**Pull request merged**
